src/structs.rs

```diff
@@ -69,6 +72,78 @@ pub trait Mergeable {
   fn merge(&self, other: Self) -> Self;
 }
 
+/// Merge two values
+fn merge_values(v1: &Value, v2: Option<&Value>) -> Value {
+  match v2 {
+    None => v1.clone(),
```

Comment:
Is this `.clone()` needed because of something something Rust memory management?

Reply:
The type passed in is a reference to a `Value` (`&Value`). The return type is a `Value` instance. `.clone()` makes a copy of the reference so it can be an instance.